src/main/java/seedu/academyassist/model/person/Person.java

```diff
@@ -25,19 +25,22 @@ public class Person {
     // Data fields
     private final Address address;
     private final Ic ic;
+    private final YearGroup yearGroup;
     private final Set<Subject> subjects = new HashSet<>();
     private final Set<Tag> tags = new HashSet<>();
 
     /**
      * Every field must be present and not null.
      */
-    public Person(Name name, Phone phone, Email email, Address address, Ic ic, Set<Subject> subjects, Set<Tag> tags) {
-        CollectionUtil.requireAllNonNull(name, phone, email, address, ic, subjects, tags);
+    public Person(Name name, Phone phone, Email email, Address address, Ic ic, YearGroup yg, Set<Subject> subjects,
+                  Set<Tag> tags) {
+        CollectionUtil.requireAllNonNull(name, phone, email, address, ic, yg, subjects, tags);
         this.name = name;
         this.phone = phone;
         this.email = email;
         this.address = address;
         this.ic = ic;
+        this.yearGroup = yg;
```

Comment:
maybe can use `this.yearGroup = yearGroup` instead of shortening it to `yg` as other parameter also uses the same naming for the parameter and the variable name for consistency.